crates/pipeline_manager/src/db/test.rs

```diff
@@ -1035,6 +1036,66 @@ pub(crate) fn limited_option_connector() -> impl Strategy<Value = Option<Connect
     })
 }
 
+/// Generate different resource configurations
+pub(crate) fn runtime_config() -> impl Strategy<Value = RuntimeConfig> {
+    any::<(
+        u16,
+        bool,
+        u64,
+        u64,
+        Option<u64>,
+        Option<u64>,
+        Option<u64>,
+        Option<u64>,
+        Option<u64>,
+    )>()
+    .prop_map(|config| RuntimeConfig {
+        workers: config.0,
+        cpu_profiler: config.1,
+        min_batch_size_records: config.2,
+        max_buffering_delay_usecs: config.3,
+        resources: ResourceConfig {
+            cpu_cores_min: config.4,
+            cpu_cores_max: config.5,
+            memory_mb_min: config.6,
+            memory_mb_max: config.7,
+            storage_mb_max: config.8,
+        },
+    })
+}
+
+/// Generate different resource configurations
+pub(crate) fn option_runtime_config() -> impl Strategy<Value = Option<RuntimeConfig>> {
```

Comment:
if you have different types you can make that configurable using the    `type Parameters = SOMEHTING;` in the Arbitrary trait impl

Reply:
This is repeating the pattern in this file where we have one strategy to generate X and another for Option<X>.